Classes/Providers/OTT/PhoenixMediaProvider.swift

```diff
@@ -0,0 +1,395 @@
+        self.playbackContextType = playbackContextType
+        return self
+    }
+
+    /// - Parameter formats: Asset's requested file formats,
+    /// According to this formats array order the sources will be ordered in the mediaEntry
+    /// According to this formats sources will be filtered when creating the mediaEntry
+    /// - Returns: Self
+    @discardableResult
+    @nonobjc public func set(formats: [String]?) -> Self {
+        self.formats = formats
+        return self
+    }
+
+    /// - Parameter formats: Asset's requested file ids,
+    /// According to this files array order the sources will be ordered in the mediaEntry
+    /// According to this ids sources will be filtered when creating the mediaEntry
+    /// - Returns: Self
+    @discardableResult
+    @nonobjc public func set(fileIds: [String]?) -> Self {
+        self.fileIds = fileIds
+        return self
+    }
+
+    /// - Parameter networkProtocol: http/https
+    /// - Returns: Self
+    @discardableResult
+    @nonobjc public func set(networkProtocol: String?) -> Self {
+        self.networkProtocol = networkProtocol
+        return self
+    }
+
+    /// - Parameter executor: executor which will be used to send request.
+    ///    default is USRExecutor
+    /// - Returns: Self
+    @discardableResult
+    @nonobjc public func set(executor: RequestExecutor?) -> Self {
+        self.executor = executor
+        return self
+    }
+
+    let defaultProtocol = "https"
+
+    /// This  object is created before loading the media in order to make sure all required attributes are set and we are ready to load
+    struct LoaderInfo {
+        var sessionProvider: SessionProvider
+        var assetId: String
+        var assetType: AssetType
+        var formats: [String]?
+        var fileIds: [String]?
+        var playbackContextType: PlaybackContextType
+        var networkProtocol: String
+        var executor: RequestExecutor
+
+    }
+
+    @objc public func loadMedia(callback: @escaping (MediaEntry?, Error?) -> Void) {
+        guard let sessionProvider = self.sessionProvider else {
+            callback(nil, PhoenixMediaProviderError.invalidInputParam(param: "sessionProvider" ).asNSError )
+            return
+        }
+        guard let assetId = self.assetId else {
+            callback(nil, PhoenixMediaProviderError.invalidInputParam(param: "assetId" ).asNSError)
+            return
+        }
+        guard let type = self.type else {
+            callback(nil, PhoenixMediaProviderError.invalidInputParam(param: "type" ).asNSError)
+            return
+        }
+        guard let contextType = self.playbackContextType else {
+            callback(nil, PhoenixMediaProviderError.invalidInputParam(param: "contextType" ).asNSError)
+            return
+        }
+
+        let pr = self.networkProtocol ?? defaultProtocol
+        var executor: RequestExecutor = USRExecutor.shared
+        if let exe = self.executor {
```

Comment:
should be remove and to executer you should write this:
`var executer: RequestExecuter = self.executer ?? USRExecutor.shared`